Sources/Overload2D.swift

```diff
@@ -0,0 +1,278 @@
+        lhs.a /= rhs; lhs.b /= rhs
+    }
+}
+
+extension CGPoint: Overload2D {
+    var a: CGFloat { get { return self.x } set { self.x = newValue } }
+    var b: CGFloat { get { return self.y } set { self.y = newValue } }
+
+    init(_ x: CGFloat, _ y: CGFloat) {
+        self.init(x: x, y: y)
+    }
+
+    static func makeTuple(_ x: CGFloat, _ y: CGFloat) -> CGPoint {
+        return CGPoint(x: x, y: y)
+    }
+
+    static func makeTuple(_ x: CGFloat, _ y: CGFloat) -> CGSize {
+        return CGSize.makeTuple(x, y)
+    }
+
+    static func makeTuple(_ x: CGFloat, _ y: CGFloat) -> CGVector {
+        return CGVector.makeTuple(x, y)
+    }
+
+    func asPoint() -> CGPoint {
+        return CGPoint(x: x, y: y)
+    }
+
+    func asSize() -> CGSize {
+        return CGSize(width: x, height: y)
+    }
+
+    func asVector() -> CGVector {
+        return CGVector(dx: x, dy: y)
+    }
+}
+
+extension CGSize: Overload2D {
+    var a: CGFloat { get { return self.width } set { self.width = newValue } }
+    var b: CGFloat { get { return self.height } set { self.height = newValue } }
+
+    init(_ width: CGFloat, _ height: CGFloat) {
+        self.init(width: width, height: height)
+    }
+
+    static func makeTuple(_ width: CGFloat, _ height: CGFloat) -> CGPoint {
+        return CGPoint.makeTuple(width, height)
+    }
+
+    static func makeTuple(_ width: CGFloat, _ height: CGFloat) -> CGSize {
+        return CGSize(width: width, height: height)
+    }
+
+    static func makeTuple(_ width: CGFloat, _ height: CGFloat) -> CGVector {
+        return CGVector.makeTuple(width, height)
+    }
+
+    func asPoint() -> CGPoint {
+        return CGPoint(x: width, y: height)
+    }
+
+    func asSize() -> CGSize {
+        return CGSize(width: width, height: height)
+    }
+
+    func asVector() -> CGVector {
+        return CGVector(dx: width, dy: height)
+    }
+}
+
+extension CGVector: Overload2D {
+    var a: CGFloat { get { return self.dx } set { self.dx = newValue } }
+    var b: CGFloat { get { return self.dy } set { self.dy = newValue } }
+
+    init(_ dx: CGFloat, _ dy: CGFloat) {
+        self.init(dx: dx, dy: dy)
+    }
+
+    static func makeTuple(_ dx: CGFloat, _ dy: CGFloat) -> CGPoint {
+        return CGPoint.makeTuple(dx, dy)
+    }
+
+    static func makeTuple(_ dx: CGFloat, _ dy: CGFloat) -> CGSize {
+        return CGSize.makeTuple(dx, dy)
+    }
+
+    static func makeTuple(_ dx: CGFloat, _ dy: CGFloat) -> CGVector {
+        return CGVector(dx: dx, dy: dy)
+    }
+
+    func asPoint() -> CGPoint {
+        return CGPoint(x: dx, y: dy)
+    }
+
+    func asSize() -> CGSize {
+        return CGSize(width: dx, height: dy)
+    }
+
+    func asVector() -> CGVector {
+        return CGVector(dx: dx, dy: dy)
+    }
+}
+
+extension CGVector {
+    func magnitude() -> CGFloat {
+        return sqrt(dx * dx + dy * dy)
+    }
+}
+
+extension CGPoint {
+    func distance(to otherPoint: CGPoint) -> CGFloat {
```

Comment:
Doesn't this calculate the magnitude of the other point regardless of `self`?

Reply:
@mats-claassen It seems that github ate my last response to you. You're right, I goofed on that part, and it made me realize I had forgotten to write unit tests for that distance function and others like it. So I closed this pull request, fixed the problems, and submitted a new one. Cheers